internal/cli/atlas/streams/instance/update.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201004/admin"
+)
+
+type UpdateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.InputOpts
+	name     string
+	provider string
+	region   string
+	store    store.StreamsUpdater
+}
+
+const (
+	updateTemplate = "Atlas Streams Processor Instance '{{.Name}}' successfully updated.\n"
+)
+
+func (opts *UpdateOpts) Run() error {
+	stream := opts.streams()
+	r, err := opts.store.UpdateStream(opts.ProjectID, opts.name, stream.DataProcessRegion)
+
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) streams() *atlasv2.StreamsTenant {
+	processor := atlasv2.NewStreamsTenant()
+	processor.Name = &opts.name
+	processor.GroupId = &opts.ProjectID
+
+	processor.DataProcessRegion = atlasv2.NewStreamsDataProcessRegionWithDefaults()
+	processor.DataProcessRegion.CloudProvider = "AWS"
```

Comment:
any reason to hard code this provider here?

Reply:
and to clarify, is the endpoint a PUT or PATCH? how flexible is it with partial updates?

Reply:
It's a PATCH, but the object we pass has both values as "required" (`ApiStreamsDataProcessRegionView`). I'm going to remove this hardcoded provider, but I will make both fields mandatory in `update`.

Reply:
Resolved in 6a37fa2a